osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/infra/jms/ws/messageprocessors/DeCoupleMbusDeviceByChannelRequestMessageProcessor.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.domain.smartmetering.infra.jms.ws.messageprocessors;
+
+import org.opensmartgridplatform.adapter.domain.smartmetering.application.services.InstallationService;
+import org.opensmartgridplatform.adapter.domain.smartmetering.infra.jms.BaseRequestMessageProcessor;
+import org.opensmartgridplatform.domain.core.valueobjects.smartmetering.DeCoupleMbusDeviceByChannelRequestData;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.infra.jms.DeviceMessageMetadata;
+import org.opensmartgridplatform.shared.infra.jms.MessageProcessorMap;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DeCoupleMbusDeviceByChannelRequestMessageProcessor extends BaseRequestMessageProcessor {
+
+    @Autowired
+    @Qualifier("domainSmartMeteringInstallationService")
+    private InstallationService installationService;
+
+    @Autowired
+    protected DeCoupleMbusDeviceByChannelRequestMessageProcessor(
```

Comment:
why protected? will this class be extended?

Reply:
all constructors of MessageProcessors in this package have protected constructors